### DISCUSSION TITLE
Post rule execution actions for multiple fields and string fields.

### DISCUSSION THREAD

**Discussion by KrupaMoliya** [microsoft/RulesEngine#484] [Q&A]
Is it possible to set/update object value OnSuccess with "OutputExpression"?
Integer field gives output by multiply, but when we use string fields it just gives true / false as a comparison result. I want to use it as assignment.  

I have 2 input Objects, let's say
stud = {"ID": 1, "Name": "Student1", "status": "NEW"}
studDtl = {"ID": 0, "Code": ""}  **// With Rule Engine, I want to set ID = 1 and Code = NewAdmission**

stud = {"ID": 2, "Name": "Student2", "status": "LEFT"}
studDtl = {"ID": 0, "Code": ""}   **// With Rule Engine, I want to set ID = 2 and Code = AlumniStudent**

I want to add Rule for different status of stud object and want to update Code in studDtl object.

    "Rules": [
      {
        "RuleName": "AlumniStudentRule",
        "SuccessEvent": "AlumniEvent",
        "ErrorMessage": "One or more adjust rules failed.",
        "ErrorType": "Error",
        "RuleExpressionType": "LambdaExpression",
        "Expression": "stud.status== \"NEW\"",
        "Actions": {
          "OnSuccess": {
            "Name": "OutputExpression",
            "Context": { **// Here I want to set studDtl.ID = stud.ID and  studDtl.Code = NewAdmission**
              "Expression": "studDtl.ID = stud.ID"
              "Expression": "studDtl.Code = \"NewAdmission\"" // If I write "Expression": "studDtl.ID * 3" it gives output as 1*3 = 3 for ruleResult.ActionResult.Output
            }
          }
        }
      }, {
        "RuleName": "NewStudentRule",
        "SuccessEvent": "NewEvent",
        "ErrorMessage": "One or more adjust rules failed.",
        "ErrorType": "Error",
        "RuleExpressionType": "LambdaExpression",
        "Expression": "stud.status== \"LEFT\"",
        "Actions": {
          "OnSuccess": { **// Here I tried Custom Action, but In Custom action I have to fetch and assign all this context that is like hardoce and will need to be updated in case of any changes** Other challenge with this is stud.ID is fetched as string "stud.ID" rather than it's value = 1
            "Name": "MyCustomAction",
            "Context": { //This is passed to the action as action context
              "studDtlCode ": "AlumniStudent",
              "studDtlID": "stud.ID"
            }
            }
          }
        }
      }

**Comment by lior-st** [comment]
@KrupaMoliya I have the exact same need. Did you find a way to achieve that functionality within actions?

**Reply by lior-st** [reply]
Here is the solution I applied: Since ActionContext does not expose a public method for getting all keys but just getting a specific context key, I added in json rules a field called "FieldsToUpdate" and in my custom action class I read this field and then I loop over field names and update the object using reflection.
